src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,183 @@
+			color: SOUL_GAUGE_COLOR.fade(0.25),
+		},
+		correctHistory: true,
+		deterministic: false,
+	}))
+
+	private shroudGauge = this.add(new CounterGauge({
+		minimum: 0,
+		maximum: SHROUD_MAX_VALUE,
+		graph: {
+			label: 'Shroud Gauge',
+			color: SHROUD_GAUGE_COLOR.fade(0.25),
+		},
+		correctHistory: true,
+		deterministic: false,
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		// soul gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_GENERATORS)), this.onSoulGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SOUL_CONSUMERS)), this.onSoulConsumption)
+
+		// shroud gauge
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_GENERATORS)), this.onShroudGeneration)
+		this.addEventHook(playerFilter.type('action').action(this.data.matchActionId(SHROUD_CONSUMERS)), this.onShroudConsumption)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onSoulGeneration(event: Events['action']) {
+		const generatedAmount = this.soulGeneratorModifiers.get(event.action)
+		if (generatedAmount) {
+			this.soulGauge.generate(generatedAmount)
+		}
+	}
+
+	private onSoulConsumption(event: Events['action']) {
+		const spentAmount = this.soulConsumptionModifiers.get(event.action)
+		if (spentAmount) {
+			this.soulGauge.spend(spentAmount)
+		}
+	}
+
+	private onShroudGeneration(event: Events['action']) {
+		const generated = this.shroudGeneratorModifiers.get(event.action)
+		if (generated) {
+			this.shroudGauge.generate(generated)
+		}
+	}
+
+	private onShroudConsumption(event: Events['action']) {
+		const spent = this.shroudConsumerModifiers.get(event.action)
+		if (spent) {
+			this.shroudGauge.spend(spent)
+		}
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.SOUL_SLICE.icon,
+			content: <Trans id="rpr.gauge.suggestions.soul-gauge-overcap.content">
+				Try to not overcap Soul gauge since it's required for shroud generation for your highest damaging skills
+				via <DataLink action="ENSHROUD"/>.
+				Pay special attention when using <DataLink action="SOUL_SLICE"/> or <DataLink action="SOUL_SCYTHE"/> as
+				they increase the gauge by <b>50</b>, make sure you have <b>50</b> or less in your gauge before using.
+			</Trans>,
+			tiers: {
+				10: SEVERITY.MINOR,
+				30: SEVERITY.MEDIUM,
+				50: SEVERITY.MAJOR,
+			},
+			value: this.soulGauge.overCap,
+			why: <Trans id="rpr.gauge.suggestions.soul-gauge-overcap.why">
+				You lost {this.soulGauge.overCap} Soul due to overcapped gauge.
+			</Trans>,
+		}))
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: this.data.actions.ENSHROUD.icon,
+			content: <Trans id="rpr.gauge.suggestions.shroud-gauge-overcap.content">
+				Try to not overcap Shroud gauge because it can lead to less usages of <DataLink action="ENSHROUD"/> over
```

Comment:
Grammar note: `fewer uses` since this is a thing that can be counted

Reply:
thanks, English is not my first language, fixed.